consultation_analyser/consultations/views/questions.py

```diff
@@ -33,3 +35,47 @@ def show(request: HttpRequest, consultation_slug: str, section_slug: str, questi
         "multiple_choice": multiple_choice,
     }
     return render(request, "show_question.html", context)
+
+
+def get_applied_filters(request: HttpRequest):
+    return {
+        "keyword": request.GET.get("keyword", ""),
+        "theme": request.GET.get("theme", "All"),
+        "opinion": request.GET.get("opinion", "All"),
+    }
+
+
+def get_filtered_responses(question: models.Question, applied_filters):
+    queryset = models.Answer.objects.filter(question=question, free_text__icontains=applied_filters["keyword"])
+    if applied_filters["theme"] != "All" and applied_filters["theme"] != "No theme":
+        queryset = queryset.filter(theme=applied_filters["theme"])
+    # TO DO: handle answers with "No theme"
+    if applied_filters["opinion"] != "All":
+        queryset = queryset.filter(multiple_choice_responses=applied_filters["opinion"])
+    return queryset
+
+
+def show_responses(request: HttpRequest, consultation_slug: str, section_slug: str, question_slug: str):
+    question = models.Question.objects.get(
+        slug=question_slug, section__slug=section_slug, section__consultation__slug=consultation_slug
+    )
+    themes_for_question = models.Theme.objects.filter(answer__question=question)
```

Comment:
In future, will be able to do `models.Theme.objects.filter(question=question)` as we're adding that field to the `Theme` model (PR still to be merged).